### PR TITLE
Multi-stage builds

### LINKED ISSUE #152: Multi-stage builds and update dockerignore
## Summary

Roll out multistage builds for design ubuntu #2412 #2310 (Web & Ubuntu Clan)

### PR DESCRIPTION
**Pull request by carkod**
## Done

Multistage builds to decrease image size based on ubuntu.com

## QA

- Build the image: `docker build --tag myimage --build-arg BUILD_ID=myfakeid .`
- Once image is built, `docker run -ti -p 8111:80 myimage`
- Check site is running smoothly at http://localhost:8111
- Check new id `myfakeid` is in the headers `curl -I http://localhost:8111`

## Issue
Fixes #152

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,18 +1,59 @@
-FROM ubuntu:xenial
+# syntax=docker/dockerfile:experimental
 
-RUN apt-get update && apt-get install --yes nginx
+# Build stage: Install ruby dependencies
+# ===
+FROM ruby:2.6 AS build-site
```
Comment (jpmartinspt) [inline Dockerfile]: Ruby 2.5 here.

File: Dockerfile [hunk 2]
```diff
@@ -1,18 +1,59 @@
...
+# Set up environment
+ENV LANG C.UTF-8
 WORKDIR /srv
-ADD _site .
+
+# Install nginx
+RUN apt-get update && apt-get install --no-install-recommends --yes nginx
+
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js requirements.txt Gemfile.lock
```
Comment (jpmartinspt) [inline Dockerfile]: remove `requirements.txt`

Comment (carkod) [reply Dockerfile]: Should we add it to the dockerignore then?

File: Dockerfile [hunk 3]
```diff
@@ -1,18 +1,58 @@
...
+# Import code, build assets and mirror list
+RUN rm -rf package.json yarn.lock .babelrc webpack.config.js Gemfile.lock
+COPY --from=build-site srv/_site .
+COPY --from=build-css srv/css css
+COPY --from=build-js srv/assets assets
+
+ARG BUILD_ID
 ADD nginx.conf /etc/nginx/sites-enabled/default
-RUN sed -i "s/~REVISION_ID~/${REVISION_ID}/" /etc/nginx/sites-enabled/default
+RUN sed -i "s/~BUILD_ID~/${BUILD_ID}/" /etc/nginx/sites-enabled/default
```
Comment (tbille) [inline Dockerfile]: Just as a reminder: https://github.com/canonical-web-and-design/microstack.run/pull/21/files#r381147043

You can update the nginx.conf file.

**Pull request merged**